wnfs-common/src/blockstore/diskblockstore.rs

```diff
@@ -0,0 +1,81 @@
+use crate::BlockStore;
+use anyhow::Result;
+use async_trait::async_trait;
+use libipld::{Cid, IpldCodec};
+use serde::{Deserialize, Serialize};
+use std::{
+    borrow::Cow,
+    path::{Path, PathBuf},
+};
+
+/// A disk-based blockstore that you can mutate.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DiskBlockStore {
+    pub path: PathBuf,
+}
+
+// -------------------------------------------------------------------------------------------------
+// Implementations
+// -------------------------------------------------------------------------------------------------
+
+impl DiskBlockStore {
+    /// Creates a new disk block store.
+    pub fn new(path: &Path) -> Self {
+        // Return the new DiskBlockStore
+        Self {
+            path: path.to_path_buf(),
+        }
+    }
+
+    pub fn erase(&self) -> Result<()> {
+        // Remove the directory
+        std::fs::remove_dir_all(&self.path)?;
+        // Return Ok status
+        Ok(())
+    }
+}
+
+impl Clone for DiskBlockStore {
+    fn clone(&self) -> Self {
+        Self::new(&self.path)
+    }
+}
+
+#[async_trait(?Send)]
+impl BlockStore for DiskBlockStore {
+    /// Stores an array of bytes in the block store.
+    async fn put_block(&self, bytes: Vec<u8>, codec: IpldCodec) -> Result<Cid> {
+        // If the parent directory doesn't already exist
+        if !self.path.exists() {
+            // Create the directories required to store the blocks
+            std::fs::create_dir_all(&self.path).unwrap();
+        }
```

Comment:
We should propagate the error here too.

```suggestion
        if !self.path.exists() {
            // Create the directories required to store the blocks
            std::fs::create_dir_all(&self.path)?;
        }
```